src/pages/workspaces/workspace/WorkspaceContainer.js

```diff
@@ -317,92 +327,24 @@ export const wrapWorkspace = ({ breadcrumbs, activeTab, title, topBarContent, sh
   const Wrapper = props => {
     const { namespace, name } = props
     const child = useRef()
-    const signal = useCancellation()
-    const [accessError, setAccessError] = useState(false)
-    const accessNotificationId = useRef()
-    const cachedWorkspace = useStore(workspaceStore)
-    const [loadingWorkspace, setLoadingWorkspace] = useState(false)
-    const workspace = cachedWorkspace && _.isEqual({ namespace, name }, _.pick(['namespace', 'name'], cachedWorkspace.workspace)) ?
-      cachedWorkspace :
-      undefined
-    const [googleProject, setGoogleProject] = useState(workspace?.workspace.googleProject)
-    const [azureContext, setAzureContext] = useState(workspace?.azureContext)
-    const [{ location, locationType }, setBucketLocation] = useState({ location: defaultLocation, locationType: locationTypes.default })
 
+    const { workspace, accessError, loadingWorkspace, storageDetails, refreshWorkspace } = useWorkspace(namespace, name)
+    const googleProject = workspace?.googleProject
+    const azureContext = workspace?.azureContext
     const prevGoogleProject = usePrevious(googleProject)
     const prevAzureContext = usePrevious(azureContext)
+
     const workspaceLoaded = !!workspace
 
     const { runtimes, refreshRuntimes, persistentDisks, appDataDisks } = useCloudEnvironmentPolling(googleProject, workspace)
     const { apps, refreshApps } = useAppPolling(googleProject, name, workspace)
     // The following if statements are necessary to support the context bar properly loading runtimes for google/azure
-    if (workspaceLoaded) {
-      if (googleProject !== prevGoogleProject || azureContext !== prevAzureContext) {
-        refreshRuntimes(isAzureWorkspace(workspace))
-        refreshApps()
-      }
+    if (workspaceLoaded && (googleProject !== prevGoogleProject || azureContext !== prevAzureContext)) {
+      refreshRuntimes(true)
```

Comment:
Cleanup discussed in an older PR-- https://github.com/DataBiosphere/terra-ui/pull/3702/files/65b6a66ad1f996447439620e45ef6c20aaf455b9#r1082910869